src/main/java/edu/wpi/first/wpilibj/networktables/NetworkTablesJNI.java

```diff
@@ -54,106 +54,283 @@ else if (System.getProperty("os.name").startsWith("Mac"))
   public static native boolean setDefaultString(String key, String defaultValue);
+  public static native boolean setDefaultString(int entry, long time, String defaultValue);
+
+  @Deprecated
   public static native boolean setDefaultRaw(String key, byte[] defaultValue);
+  public static native boolean setDefaultRaw(int entry, long time, byte[] defaultValue);
+
+  @Deprecated
   public static native boolean setDefaultBooleanArray(String key, boolean[] defaultValue);
+  public static native boolean setDefaultBooleanArray(int entry, long time, boolean[] defaultValue);
+
+  @Deprecated
   public static native boolean setDefaultDoubleArray(String key, double[] defaultValue);
+  public static native boolean setDefaultDoubleArray(int entry, long time, double[] defaultValue);
+
+  @Deprecated
   public static native boolean setDefaultStringArray(String key, String[] defaultValue);
+  public static native boolean setDefaultStringArray(int entry, long time, String[] defaultValue);
 
+  @Deprecated
   public static native void setEntryFlags(String key, int flags);
+  public static native void setEntryFlags(int entry, int flags);
+
+  @Deprecated
   public static native int getEntryFlags(String key);
+  public static native int getEntryFlags(int entry);
 
+  @Deprecated
   public static native void deleteEntry(String key);
+  public static native void deleteEntry(int entry);
+
+  @Deprecated
   public static native void deleteAllEntries();
+  public static native void deleteAllEntries(int inst);
 
+  // for backwards compatibility
+  @Deprecated
   public static native EntryInfo[] getEntries(String prefix, int types);
 
+  public static native EntryInfo getEntryInfo(int entry);
+  public static native EntryInfo[] getEntryInfo(int inst, String prefix, int types);
+
+  @Deprecated
   public static native void flush();
+  public static native void flush(int inst);
 
   @FunctionalInterface
+  @Deprecated
   public interface EntryListenerFunction {
-    void apply(int uid, String key, Object value, int flags);
+    public void apply(int entryListener, String key, Object value, int flags);
   }
+  @FunctionalInterface
+  public interface EntryListenerFunction2 {
+    public void apply(int entryListener, int entry, String key, NetworkTableValue value, int flags);
+  }
+  @Deprecated
   public static native int addEntryListener(String prefix, EntryListenerFunction listener, int flags);
-  public static native void removeEntryListener(int entryListenerUid);
+  public static native int addEntryListener(int inst, String prefix, EntryListenerFunction2 listener, int flags);
+  public static native int addEntryListener(int entry, EntryListenerFunction2 listener, int flags);
+  public static native void removeEntryListener(int entryListener);
 
   @FunctionalInterface
   public interface ConnectionListenerFunction {
-    void apply(int uid, boolean connected, ConnectionInfo conn);
+    public void apply(int connListener, boolean connected, ConnectionInfo conn);
```

Comment:
Unnecessary access modifier. Public interfaces's methods are public by default.

Also, can you add a javadoc to this method/interface, especially, what is the `connListener` parameter?